core/src/main/java/org/axonframework/common/property/DirectPropertyAccessStrategy.java

```diff
@@ -0,0 +1,28 @@
+package org.axonframework.common.property;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+import static org.axonframework.common.ReflectionUtils.fieldsOf;
+
+/**
+ * Implementation of PropertyAccessStrategy that scans class hierarchy to get public field named "property"
+ */
+public class DirectPropertyAccessStrategy extends PropertyAccessStrategy {
+
+	@Override
+	protected int getPriority() {
+		return -2048;
+	}
+
+	@Override
+	protected <T> Property<T> propertyFor(Class<? extends T> targetClass, String property) {
+		Iterable<Field> fields = fieldsOf(targetClass);
```

Comment:
Using `targetClass.getFields()` here gives you the public ones (incl. those declared on parent classes). That removes the need for checking the public modifier later on.
Did you know that you can check the modifiers using: `Modifiers.isPublic(field.getModifiers())`?